src/array.cpp

```diff
@@ -47,6 +47,19 @@ namespace sparrow
         return p_array->data_type();
     }
 
+    std::string_view array::name() const
+    {
+        ArrowSchema* schema = get_arrow_schema(const_cast<array&>(*this));
+        std::string_view res((schema->name != nullptr) ? schema->name : "");
+        return res;
+    }
```

Comment:
I suggest to not set an empty string and return an optional, and btw use the proxy implementation.
```suggestion
    std::optional<std::string_view> array::name() const
    {
        return get_arrow_proxy().name();
    }
```